modular_robot_simulation/revolve2/modular_robot_simulation/build_multi_body_systems/_body_to_multi_body_system_converter.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+from pyrr import Quaternion, Vector3
+
+from revolve2.modular_robot.body.common import Body
+from revolve2.simulation.scene import JointHinge, MultiBodySystem, Pose, RigidBody
+
+from .._modular_robot_active_hinge_key import ModularRobotActiveHingeKey
+from ._builder_params import BuilderParams
+from ._queue_item import QueueItem
+from .concrete_builder import get_builder
+
+
+class BodyToMultiBodySystemConverter:
+    """A tool to convert modular robot bodies to multi-body systems."""
+
+    _STATIC_FRICTION = 1.0
+    _DYNAMIC_FRICTION = 1.0
+
+    def convert_robot_body(
+        self, body: Body, pose: Pose, translate_z_aabb: bool
+    ) -> tuple[MultiBodySystem, dict[ModularRobotActiveHingeKey, JointHinge]]:
+        """
+        Convert a modular robot body to a multi-body system.
+
+        :param body: The body to convert.
+        :param pose: The pose to put the multi-body system in. Ownership will be taken of the pose object and it should not be reused after passing it to this function.
+        :param translate_z_aabb: Whether the robot should be translated upwards so it's T-pose axis-aligned bounding box is exactly on the ground. I.e. if the robot should be placed exactly on the ground. The pose parameters is still added afterwards.
+        :returns: The create multi-body system, a mapping from modular robot active hinges to simulation hinge joints.
+        """
+        multi_body_system = MultiBodySystem(pose=pose, is_static=False)
+        joint_mapping: dict[ModularRobotActiveHingeKey, JointHinge] = {}
+
+        rigid_body = RigidBody(
+            initial_pose=Pose(),
+            static_friction=self._STATIC_FRICTION,
+            dynamic_friction=self._DYNAMIC_FRICTION,
+            geometries=[],
+        )
+        multi_body_system.add_rigid_body(rigid_body)
+
+        queue = [
```

Comment:
I would use a deque. It is meant for this specific purpose and is significantly faster and clearer.

Something like

```
from collections import deque

queue = deque([
    QueueItem(
        body.core,
        BuilderParams((multi_body_system, joint_mapping, rigid_body, Pose()))
    )
])

while len(queue) > 0:
    current_item = queue.popleft()
    builder = get_builder(current_item.module)
    tasks = builder.build(current_item.module, current_item.builder_params)
    queue.extend(tasks)
```